src/Sylius/Component/Core/Taxation/Applicator/OrderItemUnitsTaxesApplicator.php

```diff
@@ -57,13 +57,7 @@ public function __construct(CalculatorInterface $calculator, AdjustmentFactoryIn
     public function apply(OrderInterface $order, ZoneInterface $zone)
     {
         foreach ($order->getItems() as $item) {
-            $quantity = $item->getQuantity();
-            if (0 === $quantity) {
-                continue;
-            }
```

Comment:
Actually in `Sylius\Component\Core\Taxation\Applicator\OrderItemsTaxesApplicator` there must be a check because we distribute the tax on units, but I changed the implementation to throw an exception when order item with `0` quantity is passed in order to reveal the possible bugs :)